core/src/main/java/org/infinispan/factories/EmptyConstructorNamedCacheFactory.java

```diff
@@ -122,14 +121,11 @@ public Object construct(String componentName) {
       } else if (componentName.equals(CommandsFactory.class.getName())) {
          return new CommandsFactoryImpl();
       } else if (componentName.equals(PersistenceManager.class.getName())) {
-         if (configuration.persistence().usingStores()) {
```

Comment:
There are more references to `usingStores` that need to be updated. `ActivationManagerImpl`, `PassivationManagerImpl` are just a couple as they check if stores are configured at startup.